http_types/utils.py

```diff
@@ -96,12 +96,12 @@ def delete_entries_for_serialization(data_to_be_serialized):
     for key, value in to_iter.items():
         if key == "bodyAsJson" or value is None or value == "":
             del result[key]
-        if key == "method":
+        elif key == "method":
             result[key] = result[key].value
-        if key == "protocol":
+        elif key == "protocol":
             result[key] = result[key].value
         elif isinstance(value, dict):
-            result[key] = delete_entries_for_serialization(value)
+            result[key] = fixup_entries_for_serialization(value)
```

Comment:
I wonder if these could be part of instance methods like `response.asdict()` and `response.__str__()` but I guess this is good for now!